indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -31,19 +31,25 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private final Boolean preemptive;
 
   @JsonCreator
   public TaskLock(
       @JsonProperty("groupId") String groupId,
       @JsonProperty("dataSource") String dataSource,
       @JsonProperty("interval") Interval interval,
-      @JsonProperty("version") String version
+      @JsonProperty("version") String version,
+      @JsonProperty("priority") Integer priority,
+      @JsonProperty("preemptive") Boolean preemptive
```

Comment:
hmm. after reading through most of the rest of the PR, I can't help but think that the code would be easier to follow if this property was "true" for upgraded locks and "false" for basic locks. So maybe instead of `preemptible` it should be `upgraded`? @pjain1 @himanshug @drcrallen, thoughts?

I'm happy being told to stfu on this one if people think the naming is fine the way it is